### PR TITLE
Add event logs for some startup failure cases

### PR DESCRIPTION
**Pull request by jkotalik**
For #795 and #552

### CODE REVIEW

File: src/AspNetCoreModuleV2/AspNetCore/aspnetcore_shim_config.cpp [hunk 1]
```diff
@@ -13,7 +14,7 @@ ASPNETCORE_SHIM_CONFIG::Populate(
     IHttpApplication *pHttpApplication
 )
 {
-    STACK_STRU(strHostingModel, 12);
+    STRU                            strHostingModel;
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/AspNetCore/aspnetcore_shim_config.cpp]: If someone puts an arbitrary value for the hosting model, we don't want to worry about buffer overflows.

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs [hunk 2]
```diff
@@ -280,8 +280,9 @@ private void PrepareConfig(string contentRoot, string dllRoot, int port)
             if (DeploymentParameters.PublishApplicationBeforeDeployment)
             {
                 // For published apps, prefer the content in the web.config, but update it.
-                IISDeploymentParameters.ModifyAspNetCoreSectionInWebConfig(key: "hostingModel",
+                IISDeploymentParameters.PrependAspNetCoreSectionInWebConfig(key: "hostingModel",
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs]: Need to prepend this as tests may overwrite the hostingModel section (like I do below).

File: src/AspNetCoreModuleV2/AspNetCore/aspnetcore_shim_config.cpp (hunk 1 above)
Comment (pakrym) [reply src/AspNetCoreModuleV2/AspNetCore/aspnetcore_shim_config.cpp]: STACK_STRU would reallocate AFAIK

File: src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp [hunk 3]
```diff
@@ -368,6 +371,8 @@ HOSTFXR_UTILITY::GetAbsolutePathToDotnet(
     // Only do it if no path is specified
     if (requestedPath.has_parent_path())
     {
+        WLOG_INFOF(L"Absolute path to dotnet.exe was not found at %s", requestedPath.c_str());
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp]: It's not "not found", it just wasn't an absolute path.

Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp]: So what would we have in event log in this case?

File: src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp [hunk 4]
```diff
@@ -141,14 +141,17 @@ HOSTFXR_UTILITY::GetHostFxrParameters(
...
+            EVENTLOG(hEventLog, INVALID_PROCESS_PATH, processPath.c_str(), hr);
+            return hr;
         }
 
         processPath = fullProcessPath.value();
 
         auto hostFxrPath = GetAbsolutePathToHostFxr(processPath, hEventLog);
         if (!hostFxrPath.has_value())
         {
+            hr = HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp]: You are setting hr and returning E_FAIL

File: test/IISExpress.FunctionalTests/InProcess/StartupTests.cs [hunk 5]
```diff
@@ -144,6 +146,21 @@ public static TestMatrix TestVariants
...
+            var deploymentParameters = GetBaseDeploymentParameters();
+            deploymentParameters.ModifyAspNetCoreSectionInWebConfig("hostingModel", "bogus");
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var response = await deploymentResult.RetryingHttpClient.GetAsync("HelloWorld");
+
+            Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+
+            EventLogHelpers.VerifyEventLogEvent(TestSink, "Unknown hosting model bogus. Please specify either hostingModel=\"inprocess\" or hostingModel=\"outofprocess\" in the web.config file.");
```
Comment (pakrym) [inline test/IISExpress.FunctionalTests/InProcess/StartupTests.cs]: add quotes aroung bogus

File: src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp (hunk 3 above)
Comment (jkotalik) [reply src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp]: L"Invalid or unknown processPath provided in web.config: processPath = '%s', ErrorCode = '0x%x'."

**Pull request merged**